### DISCUSSION TITLE
2-Pane-Layout in VSCode: Make Interactive Window visible when code is executed?

### DISCUSSION THREAD

**Discussion by henningsway** [quarto-dev/quarto-cli#4448] [Feature Requests]
I usually use a vertically split layout in viscose, when developing interactively.

On the left ist my Quarto document, on the right are the python and r interactive windows/consoles. Sometimes a Quarto or data or plotting preview is also in the right view.

It could be a nice option if, the interactive window would activate and become visible as soon as I execute some python or r code. Right now I usually have to select the interactive windows by mouse or keyboard shortcut, which feels a little cumbersome.

**Comment by mcanouil** [comment]
I am not sure what you are calling "Interactive Window" (all VSCode windows/panels are "interactive").

I guess you are talking about something similar to what the VSCore R extension provides:

```json
{
    "r.session.viewers.viewColumn": {
        "plot": "Beside",
        "browser": "Beside",
        "viewer": "Two",
        "pageViewer": "Two",
        "view": "Two",
        "helpPanel": "Two"
    },
}
```
<img width="1158" alt="image" src="https://user-images.githubusercontent.com/8896044/220453486-35cc1970-3fe9-4b23-9d48-fffe56369808.png">

I am pinging @jjallaire who is the one working on the VSCode extension (<https://github.com/quarto-dev/quarto>).

**Comment by jjallaire** [comment]
Currently we use the APIs of the R and Python extensions to execute code interactively. It seems logical that they would activate automatically but it seems like they don't?

**Comment by henningsway** [comment]
I guess, I'm still look for the best kind of window management when developing interactively with Quarto and Python/R. :)

Whenever I render the Quarto document, the Preview tab comes to the foreground, but when I execute some python or R code right after, the preview stays active and the focus doesn't jump back to the interactive python or r window, where the code is executed.

To me it seems, as if an option to "autofocus" and change the focused editor-tabs, depending on where the evaluation of my code takes place could be a nice addition.

**Comment by mcanouil** [comment]
The matter is not really about "is it something to add or to be done or not?".
We (or at least I) can't understand what's your issue is exactly, i.e., you do not describe a workflow to reproduce, nor provide a video showing your issue.
Without a more detailed description of your issue, it's nearly impossible to do something about it.

**Comment by mcanouil** [comment]
Closing this.
For further requests on this, please do so on <https://github.com/quarto-dev/quarto>.
